contracts/EverlongStrategy.sol

```diff
@@ -219,6 +221,31 @@ contract EverlongStrategy is BaseStrategy {
         return;
     }
 
+    /// @dev Withdraw function that can be called after the vault is shut down.
+    ///      Takes all longs controlled by the strategy and transfers them to
+    ///      the emergency admin address.
+    /// @param . Amount of assets to withdraw. This is ignored to reduce the
+    ///          likelihood of reverts.
+    function _emergencyWithdraw(uint256) internal override {
+        IEverlongStrategy.EverlongPosition memory position;
+        while (!_portfolio.isEmpty()) {
```

Comment:
there might be a lot of open positions and closing them is not cheap. consider just closing positions until at least `amount` parameter is reached.